spec/controllers/recordings_controller_spec.rb

```diff
@@ -53,6 +53,102 @@
+      }
+    end
+
+    let(:pre_update_recordings_list) do
+      { returncode: true,
+        recordings: [{ recordID: 'd1784817183960eea1c2e5738eb8c62681f42e23-1654674116107',
+                       meetingID: 'jpykqjw7jvbe3s7dgxdsdl5dkmajusro62qfmzff',
+                       name: 'Newer',
+                       published: true,
+                       protected: 'false',
+                       state: 'published',
+                       startTime: 'Wed, 08 Jun 2022 08:41:56 +0100'.to_datetime,
+                       endTime: 'Wed, 08 Jun 2022 08:42:38 +0100'.to_datetime,
+                       participants: '1',
+                       metadata: { 'bbb-origin': 'greenlight', 'bbb-origin-version': '3', isBreakout: 'false',
+                                   'bn-presenter-name': 'Presenter' },
+                       playback: { format: { type: 'presentation',
+                                             url: 'https://blindsidenetworks.com/',
+                                             length: 1 } } }],
+        messageKey: '',
+        message: '' }
+    end
+
+    before do
+      room = create(:room, user:)
+      room.meeting_id = 'jpykqjw7jvbe3s7dgxdsdl5dkmajusro62qfmzff'
+      room.save
+
+      fake_bbb_api = Class.new(BigBlueButtonApi) do
+        def update_recordings(*_args)
+          BigBlueButtonApi::RECORDINGS_LIST[:recordings] = [
+            { recordID: 'd1784817183960eea1c2e5738eb8c62681f42e23-1654674116107',
+              meetingID: 'jpykqjw7jvbe3s7dgxdsdl5dkmajusro62qfmzff',
+              name: 'Newer',
+              published: true,
+              protected: 'false',
+              state: 'published',
+              startTime: 'Wed, 08 Jun 2022 08:41:56 +0100'.to_datetime,
+              endTime: 'Wed, 08 Jun 2022 08:42:38 +0100'.to_datetime,
+              participants: '1',
+              metadata: { 'bbb-origin': 'greenlight', 'bbb-origin-version': '3', isBreakout: 'false',
+                          'bn-presenter-name': 'Presenter', 'recording-name': 'Older' },
+              playback: { format: { type: 'presentation',
+                                    url: 'https://blindsidenetworks.com/',
+                                    length: 1 } } }
+          ]
+        end
+
+        def get_recordings(*_args)
+          BigBlueButtonApi::RECORDINGS_LIST
+        end
+      end
+
+      stub_const('BigBlueButtonApi', fake_bbb_api)
+      stub_const('BigBlueButtonApi::RECORDINGS_LIST', pre_update_recordings_list)
+    end
+
+    it 'updates the recordings name with valid params returning :ok status code' do
+      expect { get :resync }.to change(Recording, :count).from(0).to(BigBlueButtonApi::RECORDINGS_LIST[:recordings].size)
+      recording_id = Recording.pluck(:record_id).sample
+      post :update, params: { recording: recording_update_params, id: recording_id }
+      expect(response).to have_http_status(:ok)
+      get :resync
+      recording = Recording.find_by record_id: recording_id
```

Comment:
This is definitely way too complex. I think you should review the other tests and approach them in a similar way.

Reply:
Totally agree, after fallowing the new design this became absurd.